pkg/vm/engine/disttae/reader.go

```diff
@@ -40,8 +40,9 @@ func (r *blockReader) Read(cols []string, expr *plan.Expr, m *mpool.MPool) (*bat
 		return nil, nil
 	}
 	defer func() { r.blks = r.blks[1:] }()
-	return blockio.BlockRead(r.ctx, cols, r.tableDef, r.blks[0].Info.MetaLoc,
-		r.blks[0].Info.DeltaLoc, r.ts, r.fs, m)
+	info := &r.blks[0].Info
+	return blockio.BlockRead(r.ctx, info.BlockID, info.SegmentID, cols, r.tableDef,
```

Comment:
Why not just accept catalog.BlockInfo directly?

Reply:
OK, i will change it